src/cascadia/TerminalApp/CascadiaSettingsSerialization.cpp

```diff
@@ -83,29 +121,78 @@ std::unique_ptr<CascadiaSettings> CascadiaSettings::LoadDefaults()
 
     return resultPtr;
 }
 
 // Method Description:
-// - Attempts to read the given data as a string of JSON, parse that JSON, and
-//   then layer the settings from that JSON object on our current settings. See
-//   CascadiaSettings::LayerJson for detauls on how the layering works.
+// - Runs each of the configured dynamic profile generators (DPGs). Adds
+//   profiles from any DPGs that ran to the end of our list of profiles.
+// - Uses the Json::Value _userSettings to check which DPGs should not be run.
+//   If the user settings has any namespaces in the "disabledProfileSources"
+//   property, we'll ensure that any DPGs with a matching namespace _don't_ run.
+// Arguments:
+// - <none>
+// Return Value:
+// - <none>
+void CascadiaSettings::_LoadDynamicProfiles()
+{
+    std::unordered_set<std::wstring> ignoredNamespaces;
+    const auto disabledProfileSources = CascadiaSettings::_GetDisabledProfileSourcesJsonObject(_userSettings);
+    if (disabledProfileSources.isArray())
+    {
+        for (const auto& ns : disabledProfileSources)
+        {
+            ignoredNamespaces.emplace(GetWstringFromJson(ns));
+        }
+    }
+
+    const GUID nullGuid{ 0 };
+    for (auto& generator : _profileGenerators)
+    {
+        const std::wstring generatorNamespace{ generator->GetNamespace() };
+
+        if (ignoredNamespaces.find(generatorNamespace) != ignoredNamespaces.end())
+        {
+            // namespace should be ignored
+        }
+        else
+        {
+            auto profiles = generator->GenerateProfiles();
+            for (auto& profile : profiles)
+            {
+                // If the profile did not have a GUID when it was generated,
+                // we'll synthesize a GUID for it in _ValidateProfilesHaveGuid
+                profile.SetSource(generatorNamespace);
+
+                _profiles.emplace_back(profile);
+            }
+        }
+    }
+}
+
+// Method Description:
+// - Attempts to read the given data as a string of JSON and parse that JSON
+//   into a Json::Value.
 // - Will ignore leading UTF-8 BOMs.
 // - Additionally, will store the parsed JSON in this object, as either our
 //   _defaultSettings or our _userSettings, depending on isDefaultSettings.
+// - Does _not_ apply the json onto our current settings. Callers should make
+//   sure to call LayerJson to ensure the settings are applied.
 // Arguments:
 // - fileData: the string to parse as JSON data
 // - isDefaultSettings: if true, we should store the parsed JSON as our
 //   defaultSettings. Otherwise, we'll store the parsed JSON as our user
 //   settings.
 // Return Value:
 // - <none>
-void CascadiaSettings::_LayerJsonString(std::string_view fileData, const bool isDefaultSettings)
+void CascadiaSettings::_ParseJsonString(std::string_view fileData, const bool isDefaultSettings)
 {
     // Ignore UTF-8 BOM
     auto actualDataStart = fileData.data();
+    const auto actualDataEnd = fileData.data() + fileData.size();

```

Comment:
nit: is this different from `fileData.end()`?

Reply:
The type certainly is. Intellisense is borked ATM but I had a great deal of trouble using `.end()` here.